notebooks/image_models/solutions/diffusion_model.ipynb

```diff
@@ -0,0 +1,723 @@
+{
```

Comment:
<p>Since this code block is very large and it includes important concepts, I think it's better to explain the purpose and overview of each function first, and then elaborate the <code>train_step</code> function.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/347/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>